src/main/java/school/faang/user_service/service/RecommendationService.java

```diff
@@ -0,0 +1,121 @@
+import school.faang.user_service.repository.recommendation.RecommendationRepository;
+import school.faang.user_service.repository.recommendation.SkillOfferRepository;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+
+@Component
+@RequiredArgsConstructor
+public class RecommendationService {
+    private final static int PERIOD_TO_ADD_NEW_RECOMMENDATION = 6;
+    private final RecommendationRepository recommendationRepository;
+    private final SkillOfferRepository skillOfferRepository;
+    private final SkillRepository skillRepository;
+    private final RecommendationMapper recommendationMapper;
+
+    public RecommendationDto create(RecommendationDto recommendation) {
+        validateRecommendationForPeriod(recommendation);
+        validateSkillsInSystem(recommendation);
+
+        List<Long> skillIds = getSkillIds(recommendation);
+        Long recommendationId = recommendationRepository
+                .create(recommendation.getAuthorId(), recommendation.getReceiverId(), recommendation.getContent());
+        saveSkillOffers(recommendationId, skillIds);
+
+        return recommendation;
+    }
+
+    public RecommendationDto update(RecommendationDto recommendation) {
+        validateRecommendationForPeriod(recommendation);
+        validateSkillsInSystem(recommendation);
+
+        recommendationRepository
+                .update(recommendation.getAuthorId(), recommendation.getReceiverId(), recommendation.getContent());
+        skillOfferRepository.deleteAllByRecommendationId(recommendation.getId());
+        List<Long> skillIds = getSkillIds(recommendation);
+        for (Long skillId : skillIds) {
+            skillOfferRepository.create(skillId, recommendation.getId());
+        }
+
+        return recommendation;
+    }
+
+    public void delete(Long recommendationId) {
+        recommendationRepository.deleteById(recommendationId);
+    }
+
+    public List<RecommendationDto> getAllUserRecommendations(long receiverId) {
+        Page<Recommendation> allByReceiverId = recommendationRepository
+                .findAllByReceiverId(receiverId, Pageable.unpaged());
+        List<Recommendation> recommendations = allByReceiverId.getContent();
+
+        return recommendationMapper.mapToDtoList(recommendations);
+    }
+
+    public List<RecommendationDto> getAllGivenRecommendations(long authorId) {
+        Page<Recommendation> allByAuthorId = recommendationRepository
+                .findAllByAuthorId(authorId, Pageable.unpaged());
+        List<Recommendation> recommendations = allByAuthorId.getContent();
+
+        return recommendationMapper.mapToDtoList(recommendations);
+    }
+
+    private void validateRecommendationForPeriod(RecommendationDto recommendation) {
+        Optional<LocalDateTime> lastRecommendationDate;
+        lastRecommendationDate = getLastRecommendationDate(recommendation.getAuthorId(), recommendation.getReceiverId());
```

Comment:
```suggestion
        Optional<LocalDateTime> lastRecommendationDate = getLastRecommendationDate(recommendation.getAuthorId(), recommendation.getReceiverId());
```
Можно сразу в переменную присвоить